dbms/src/Operators/AutoPassThroughHashAggHelper.cpp

```diff
@@ -0,0 +1,496 @@
+                return [col_index](const Block & block) {                                                              \
+                    return genPassThroughColumnForSumNumber<FROM_NATIVE_TYPE, TO_NATIVE_TYPE, false>(                  \
+                        col_index,                                                                                     \
+                        block);                                                                                        \
+                };                                                                                                     \
+        }                                                                                                              \
+    } while (0);
+
+#define INNER_UInt8 FOR_NUMBER_TYPES_INNER(UInt8, DISPATCH_NUMBER_INNER)
+#define INNER_UInt16 FOR_NUMBER_TYPES_INNER(UInt16, DISPATCH_NUMBER_INNER)
+#define INNER_UInt32 FOR_NUMBER_TYPES_INNER(UInt32, DISPATCH_NUMBER_INNER)
+#define INNER_UInt64 FOR_NUMBER_TYPES_INNER(UInt64, DISPATCH_NUMBER_INNER)
+#define INNER_Int8 FOR_NUMBER_TYPES_INNER(Int8, DISPATCH_NUMBER_INNER)
+#define INNER_Int16 FOR_NUMBER_TYPES_INNER(Int16, DISPATCH_NUMBER_INNER)
+#define INNER_Int32 FOR_NUMBER_TYPES_INNER(Int32, DISPATCH_NUMBER_INNER)
+#define INNER_Int64 FOR_NUMBER_TYPES_INNER(Int64, DISPATCH_NUMBER_INNER)
+#define INNER_Float32 FOR_NUMBER_TYPES_INNER(Float32, DISPATCH_NUMBER_INNER)
+#define INNER_Float64 FOR_NUMBER_TYPES_INNER(Float64, DISPATCH_NUMBER_INNER)
+
+#define DISPATCH_NUMBER_OUTER(FROM_NATIVE_TYPE)                                            \
+    do                                                                                     \
+    {                                                                                      \
+        if (checkDataType<DataTypeNumber<FROM_NATIVE_TYPE>>(from_type.get())) /* NOLINT */ \
+        {                                                                                  \
+            INNER_##FROM_NATIVE_TYPE                                                       \
+        }                                                                                  \
+    } while (0);
+
+#define FOR_NUMBER_TYPES_OUTER(M) \
+    M(UInt8)                      \
+    M(UInt16)                     \
+    M(UInt32)                     \
+    M(UInt64)                     \
+    M(Int8)                       \
+    M(Int16)                      \
+    M(Int32)                      \
+    M(Int64)                      \
+    M(Float32)                    \
+    M(Float64)
+
+    FOR_NUMBER_TYPES_OUTER(DISPATCH_NUMBER_OUTER);
+
+#undef INNER_UInt8
+#undef INNER_UInt16
+#undef INNER_UInt32
+#undef INNER_UInt64
+#undef INNER_Int8
+#undef INNER_Int16
+#undef INNER_Int32
+#undef INNER_Int64
+#undef INNER_Float32
+#undef INNER_Float64
+#undef DISPATCH_NUMBER_INNER
+#undef DISPATCH_NUMBER_OUTER
+#undef FOR_NUMBER_TYPES_INNER
+#undef FOR_NUMBER_TYPES_OUTER
```

Comment:
I think no need to care integer type to decimal type. Because the sum agg func implementation didn't care. Only decimal to decimal and integer to integer need to be consider.

Reply:
check https://github.com/pingcap/tiflash/blob/334d9b97d94f2402ae2c2446fe79ff1a3c467162/dbms/src/AggregateFunctions/AggregateFunctionSum.cpp#L85-L113

Reply:
you mean when user input is sum(long), TiDB will convert it to sum(cast(long to decimal))?

Reply:
Yes

Reply:
following is the example(explain analyze select /*+ mpp_2phase_agg() */ sum(c1) from tt group by c2; c1 and c2 are both int type) 
![i0L2JBX8Ec](https://github.com/user-attachments/assets/15779947-33bc-4cc0-863b-6fb2091dd5fa)